nemo/utils/callbacks/dist_ckpt_io.py

```diff
@@ -59,7 +70,7 @@ def load_checkpoint(
         if map_location is not None:
             raise ValueError('DistributedCheckpointIO doesnt handle map_location argument')
 
-        if self.save_ckpt_format == 'zarr':
+        if self.save_ckpt_format == 'zarr' and self.load_directly_on_device:
```

Comment:
Hi, will this work? Why not pass self.load_directly_on_device as an arg to TensorStoreLoadShardedStrategy?

Reply:
It's the same behavior, and this one approach allows changing the underlying default strategy (if TensorStoreLoadShardedStrategy is not used explicitly)